pkg/controller/mattermostcluster/mattermostcluster_controller.go

```diff
@@ -100,54 +110,524 @@ func (r *ReconcileMattermostCluster) Reconcile(request reconcile.Request) (recon
+		mattermost.Spec.Image = constants.DefaultMattermostImage
+	}
+
+	if mattermost.Spec.Replicas == 0 {
+		reqLogger.Info("Setting default Mattermost replicas: " + strconv.Itoa(constants.DefaultAmountOfPods))
+		mattermost.Spec.Replicas = constants.DefaultAmountOfPods
+	}
+
+	if len(mattermost.Spec.DatabaseType.Type) == 0 {
+		reqLogger.Info("Setting default Mattermost database type: " + constants.DefaultMattermostDatabaseType)
+		mattermost.Spec.DatabaseType.Type = constants.DefaultMattermostDatabaseType
+	}
+
+	return nil
+}
+
+func (r *ReconcileMattermostCluster) checkDBServiceAccount(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) error {
+
+	// Service Account for Mysql Oracle Operator
+	serviceAccountName := "mysql-agent"
+	serviceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:    map[string]string{constants.ClusterLabel: mattermost.Name},
+			Name:      serviceAccountName,
+			Namespace: mattermost.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(mattermost, schema.GroupVersionKind{
+					Group:   mattermostv1alpha1.SchemeGroupVersion.Group,
+					Version: mattermostv1alpha1.SchemeGroupVersion.Version,
+					Kind:    "MattermostCluster",
+				}),
+			},
+		},
+	}
+
+	foundServiceAccount := &corev1.ServiceAccount{}
+	errGet := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: mattermost.Namespace}, foundServiceAccount)
+	if errGet != nil && errors.IsNotFound(errGet) {
+		reqLogger.Info("MattermostCluster Database Service account not found, will deploy one.", "Err", errGet.Error())
+		reqLogger.Info("Creating Service Account")
+		if err := r.client.Create(context.TODO(), serviceAccount); err != nil {
+			reqLogger.Info("Error creating Service Account", "Error", err.Error())
+			return err
+		}
+		// TODO add logic to wait for the completion
+		reqLogger.Info("Creating Database Service Account completed")
+		if err := controllerutil.SetControllerReference(mattermost, serviceAccount, r.scheme); err != nil {
+			return err
+		}
+	} else if errGet != nil {
+		reqLogger.Error(errGet, "MattermostCluster Database Service Account")
+		return errGet
 	}
-	return &corev1.Pod{
+
+	// TODO check how to update without creating new tokens
+	// Set MattermostService instance as the owner and controller
+	// if !reflect.DeepEqual(serviceAccount, foundServiceAccount) {
+	// 	foundServiceAccount = serviceAccount
+	// 	reqLogger.Info("Updating Service Account", serviceAccount.Namespace, serviceAccount.Name)
+	// 	err = r.client.Update(context.TODO(), foundServiceAccount)
+	// 	if err != nil {
+	// 		return err
+	// 	}
+	// 	_ = controllerutil.SetControllerReference(mattermost, foundServiceAccount, r.scheme)
+	// }
+
+	return nil
+}
+
+func (r *ReconcileMattermostCluster) checkDBRoleBinding(mattermost *mattermostv1alpha1.MattermostCluster, reqLogger logr.Logger) error {
```

Comment:
Similar suggestion of splitting into smaller functions as above with `checkDBServiceAccount`. We should do this for all similar methods